pkg/controller/manager.go

```diff
@@ -41,17 +41,22 @@ func GetGlobalStatus() models.ControllerStatuses {
 	return globalStatus.GetStatusModel()
 }
 
-// UpdateController installs or updates a controller in the manager. A
-// controller is identified by its name. If a controller with the name already
-// exists, the controller will be shut down and replaced with the provided
-// controller. Updating a controller will cause the DoFunc to be run
-// immediately regardless of any previous conditions. It will also cause any
-// statistics to be reset.
-func (m *Manager) UpdateController(name string, params ControllerParams) {
-	m.updateController(name, params)
+// UpdateController installs or updates a controller in the
+// manager. A controller is primarily identified by its name.
+// If a controller with the name already exists, the controller
+// will be shut down and replaced with the provided controller.
+//
+// A controller's group is used for aggregate metrics collection.
+// The group.Name must NOT be dynamically generated from a
+// resource identifier, in order to limit metrics cardinality.
+//
+// Updating a controller will cause the DoFunc to be run immediately regardless
+// of any previous conditions. It will also cause any statistics to be reset.
+func (m *Manager) UpdateController(group ControllerGroup, name string, params ControllerParams) {
```

Comment:
Rather than changing the signature of `UpdateController`, what about adding the option to set a group for an entire Manager?

If that doesn't make sense / needs to be overridden, then the ControllerGroup should be a member in ControllerParams